### PR TITLE
Update ConnectorIndex

### PR DESCRIPTION
**Pull request by wangch079**
This PR made those changes:
1. move the query-building logic to `supported_connectors` method, as it's only used here.
2. Use method `supported_connectors` directly, instead of calling `build_docs_query` and then `get_all_docs`
3. change `update_filtering_validation` to only update `filtering` instead of the whole document.
4. add some wrapping methods to `ESIndex`
5. rename all `connectors_ids` to `connector_ids`

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -107,31 +106,34 @@ async def save(self, connector):
...
+        await self.update(doc_id=doc_id, doc={"last_seen": iso_utc()})
+
     async def update_filtering_validation(
         self, connector, validation_result, validation_target=ValidationTarget.ACTIVE
     ):
-        doc_to_update = deepcopy(connector.doc_source)
+        filtering = connector.filtering.to_list()
 
-        for filter_ in doc_to_update.get("filtering", []):
+        for filter_ in filtering:
```
Comment (vidok) [inline connectors/byoc.py:117]: Just wonder if know why `_` in `filter_` is needed?

Comment (wangch079) [reply connectors/byoc.py:117]: If you rename it to `filter`, IDE will complain:
```
Shadows built-in name 'filter' 
```

Comment (vidok) [reply connectors/byoc.py:117]: That's weird because I don't see any other `filter` names it shadows.

Comment (tarekziade) [reply connectors/byoc.py:117]: `filter` is a built-in class in Python itself -- so adding `_` is the right pattern

File: connectors/byoc.py [hunk 2]
```diff
@@ -303,6 +296,9 @@ def get_filter(self, filter_state="active", domain=DEFAULT_DOMAIN):
             Filter(),
         )
 
+    def to_list(self):
```
[the remark this answers is not in the document]
Comment (artem-shelkovnikov) [reply connectors/byoc.py:299]: I also feel like `to_list()` method is not adding much value here - `list(connector.filtering)` looks more or less nice to me.

File: connectors/byoc.py [hunk 3]
```diff
@@ -84,11 +84,10 @@ class DataSourceError(Exception):
 class ConnectorIndex(ESIndex):
     def __init__(self, elastic_config):
         logger.debug(f"ConnectorIndex connecting to {elastic_config['host']}")
-        # initilize ESIndex instance
+        # initialize ESIndex instance
```
Comment (artem-shelkovnikov) [inline connectors/byoc.py:87]: While I think it's okay to keep it in this PR, ideally renames/typo fixes should go in a different PR, especially if there are many of them - they just distract from the actual change IMO.

File: connectors/byoc.py [hunk 4]
```diff
@@ -107,31 +106,34 @@ async def save(self, connector):
...
-            index=CONNECTORS_INDEX,
+            index=self.index_name,
             id=connector.id,
-            doc=doc_to_update,
+            doc={"filtering": filtering},
             retry_on_conflict=RETRY_ON_CONFLICT,
         )
 
-    def build_docs_query(self, native_service_types=None, connectors_ids=None):
+    async def supported_connectors(self, native_service_types=None, connector_ids=None):
```
Comment (artem-shelkovnikov) [inline connectors/byoc.py:130]: cc @vidok - I know you added the `build_docs_query` originally and passed the query inside a different method. What was the goal of original design and how do you feel about this change?

I'm not against it - it looks okay, but want to understand a bit more about it

File: connectors/tests/test_byoc.py [hunk 5]
```diff
@@ -337,7 +331,86 @@ async def test_heartbeat(mock_responses, patch_logger):
...
-async def test_connectors_get_list(mock_responses):
+async def test_connectors_get_list_no_connector(mock_responses):
+    config = {"host": "http://nowhere.com:9200", "user": "tarek", "password": "blah"}
+    connectors = ConnectorIndex(config)
+    conns = []
+    async for connector in connectors.supported_connectors():
+        conns.append(connector)
+
+    assert len(conns) == 0
+    await connectors.close()
```
Comment (artem-shelkovnikov) [inline connectors/tests/test_byoc.py:342]:
```suggestion
    await connectors.close()
    
    assert len(conns) == 0
```

I'd just flip them around cause asserts should be last in each individual test

**Review by artem-shelkovnikov — APPROVED** [review]
I think change looks okay, just have a few nits for it, so it's okay to merge and refactor more IMO. If you are strongly against it, please do the "Request changes" review on it.

File: connectors/byoc.py (hunk 4 above)
Comment (wangch079) [reply connectors/byoc.py:130]: Initially, I wanted to rename `build_docs_query` to `supported_connectors_query`, because `build_docs_query` is like a general purpose query building method, but it's not.

Then I realized even `supported_connectors_query` was only used by the method `supported_connectors`.

So In `supported_connectors`, it simply does 2 things, build the query, and then call `get_all_docs`.

File: connectors/byoc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wangch079) [reply connectors/byoc.py:299]: > `list(connector.filtering)` looks more or less nice to me.

It's not `list(connector.filtering)`, but `list(connector.filtering.filtering)`
